### PR TITLE
fix in hosts param crud test

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
assertion error

### Solution
reacting to change in param object

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_parameter_crud

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7672
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_parameter_crud --external-logging
Test Result : ================== 1 passed, 9 warnings in 668.81s (0:11:08) ===================
```

**Review by vijaysawant — COMMENTED** [review]
Why we are changing the older way of assertion? what is return value of `host['parameters']` ?
I would assert with exact `name` & `value` rather asserting with list type object.
```
assert name == host['parameters']['name']
assert value == host['parameters']['value']
```

File: tests/foreman/cli/test_host.py [hunk 1]
```diff
@@ -1105,18 +1105,19 @@ def test_positive_parameter_crud(function_host, target_sat):
         {'host-id': function_host['id'], 'name': name, 'value': value}
     )
     host = target_sat.cli.Host.info({'id': function_host['id']})
-    assert name in host['parameters']
-    assert value == host['parameters'][name]
+    assert name in [param["name"] for param in host['parameters']]
+    assert value in [param["value"] for param in host['parameters']]
```
Comment (vijaysawant) [inline tests/foreman/cli/test_host.py:1109]: If we are going to continue with these new changes then I would suggest to use single quote for string to maintain consistant pattern
```suggestion
    assert name in [param['name'] for param in host['parameters']]
    assert value in [param['value'] for param in host['parameters']]
```

File: tests/foreman/cli/test_host.py [hunk 2]
```diff
@@ -1105,18 +1105,19 @@ def test_positive_parameter_crud(function_host, target_sat):
...
 
     new_value = valid_data_list()[name]
     target_sat.cli.Host.set_parameter({'host-id': host['id'], 'name': name, 'value': new_value})
     host = target_sat.cli.Host.info({'id': host['id']})
-    assert name in host['parameters']
-    assert new_value == host['parameters'][name]
+
+    assert name in [param["name"] for param in host['parameters']]
+    assert new_value in [param["value"] for param in host['parameters']]
```
Comment (vijaysawant) [inline tests/foreman/cli/test_host.py:1116]:
```suggestion
    assert name in [param['name'] for param in host['parameters']]
    assert new_value in [param['value'] for param in host['parameters']]
```

File: tests/foreman/cli/test_host.py [hunk 3]
```diff
@@ -1105,18 +1105,19 @@ def test_positive_parameter_crud(function_host, target_sat):
...
 
     target_sat.cli.Host.delete_parameter({'host-id': host['id'], 'name': name})
     host = target_sat.cli.Host.info({'id': host['id']})
-    assert name not in host['parameters']
+    assert name not in [param["name"] for param in host['parameters']]
```
Comment (vijaysawant) [inline tests/foreman/cli/test_host.py:1120]:
```suggestion
    assert name not in [param['name'] for param in host['parameters']]
```

File: tests/foreman/cli/test_host.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/cli/test_host.py:1109]: Can we merge these assertions like "(name,value) in [(param['name'], param['value']) for param in host['parameters']]"

**Review by shweta83 — CHANGES_REQUESTED** [review]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_parameter_crud

**Comment by pondrejk** [comment]
> Why we are changing the older way of assertion? what is return value of `host['parameters']` ? I would assert with exact `name` & `value` rather asserting with list type object.
> 
> ```
> assert name == host['parameters']['name']
> assert value == host['parameters']['value']
> `, 
host['parameters'] is a list of dictionaries, something like [{name: 'a', value:'b'}]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7692
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_parameter_crud --external-logging
Test Result : ================== 1 passed, 9 warnings in 645.04s (0:10:45) ===================
```

**Comment by vijaysawant** [comment]
> > Why we are changing the older way of assertion? what is return value of `host['parameters']` ? I would assert with exact `name` & `value` rather asserting with list type object.
> > ```
> > assert name == host['parameters']['name']
> > assert value == host['parameters']['value']
> > `, 
> > ```
> 
> host['parameters'] is a list of dictionaries, something like [{name: 'a', value:'b'}]

we can do something like this, can you please check and update?
```
for key, value in host['parameters'].items():
    assert (name, value) == (key, value)
```

**Review by vijaysawant — APPROVED** [review]
Thanks for work @pondrejk 
Ack

**Review by lhellebr — APPROVED** [review]

**Review by shweta83 — APPROVED** [review]
Ack.

**Pull request merged**